src/main/java/org/elasticsearch/search/dfs/DfsPhase.java

```diff
@@ -61,7 +61,12 @@ public void execute(SearchContext context) {
                 context.updateRewriteQuery(context.searcher().rewrite(context.query()));
             }
 
-            context.query().extractTerms(new DelegateSet(termsSet));
```

Comment:
extractTerms seems broken here: since TermsFilter is not exposed yet as a query, it means someone has implemented extractTerms incorrectly.

This is traditionally used by highlighters etc, and traditionally non-scoring clauses like MUST_NOT were ignored.

There is at least a bug in BooleanQuery.java, it only ignores prohibited clauses. But this should be changed to ignore "non-scoring" clauses. And FilteredQuery or whoever else should not call it on the Filter side.